innovation.game.php

```diff
@@ -10486,6 +10505,58 @@ function stInterInteractionStep() {
                     }
                     break;
                 
+				// id 123, Artifacts age 1: Ark of the covenant
+                case "123N1A":
+					$players = self::loadPlayersBasicInfos();
+					
+					if ($n > 0)
+					{ // Unsaid rule: the player must have at least one card to show from his hand, else, the effect can't continue
+                        $color = self::getGameStateValue('color_last_selected');
+                            
+                        foreach($players as $all_player_id => $player) 
+						{
+                            // TODO : is there a better way to get all top cards?
```

Comment:
Can you write a new `getTopCardsOnBoard` function? It will be quite similar to `getTopCardOnBoard` except we we drop the bits which restrict the query to a specific color, and you'll also need to use `getCollectionFromDb` instead of `getObjectFromDB` since we are grabbing an array of elements instead of a single element.

I just ran the following query on the database and it works:

```
                SELECT
                    *
                FROM
                    card
                WHERE
                    card.owner = {player_id} AND
                    card.location = 'board' AND
                    card.position = (
                        SELECT
                            MAX(position) AS position
                        FROM
                            card
                        WHERE
                            owner = {player_id} AND
                            location = 'board' AND
                            color = card.color
                    )
```

Reply:
I'm not sure you want to trust me with SQL...heh.  I can give it a try.

Reply:
Haha I already wrote out the tested SQL query so it's just a matter of putting it in a PHP function!